iothub/device/src/Transport/ProtocolRoutingDelegatingHandler.cs

```diff
@@ -83,22 +89,20 @@ public override async Task OpenAsync(CancellationToken cancellationToken)
 
         private void CreateNewTransportIfNotReady()
         {
-            if (InnerHandler == null || !InnerHandler.IsUsable)
+            if (!(InnerHandler?.IsUsable ?? false))
             {
                 CreateNewTransportHandler();
             }
         }
 
         private void CreateNewTransportHandler()
         {
-            if (InnerHandler != null)
-            {
-                InnerHandler.Dispose();
-                InnerHandler = null;
-            }
+            IDelegatingHandler innerHandler = InnerHandler;
 
             // Ask the ContinuationFactory to attach the proper handler given the Context's ITransportSettings.
             InnerHandler = ContinuationFactory(Context, null);
+
+            innerHandler?.Dispose();
```

Comment:
Please add a comment explaining why CreateNewTransportHandler() during dispose is OK in `WaitForTransportClosedAsync`.

Reply:
done